### DISCUSSION TITLE
How to use UCX in multiple processes

### DISCUSSION THREAD

**Discussion by hanzhihua112** [openucx/ucx#7589] [Q&A] [answered]
I'm using UCX. The server requirements are as follows:
1. The parent process listens on the port and blocks in the waiting connection position
2. If a client connection is received, create a child process to process the message
3. The child process will communicate with the client separately
 
Can you give us some guidance?

**Comment by evgeny-leksikov** [comment, marked as answer]
@hanzhihua112 

UCX does not support handling connection request from child process. Probably something from below can meet your requrements:
1. create new worker in scope of the same process (it could be separate thread) and handle connection request and messaging there 
2. another pair of endpoints for communication between client and child-server:
     - create connection on listen process (accept connection request)
     - create child process with separate UCX context/worker
     - exchange worker addresses using origin server
     - close origin connections
     - create another pair of endpoints (on client and server-child) connected to each other by worker address

**Comment by shamisp** [comment]
I will comment that this is not a UCX limitation but the typical limitation within RDMA NICs.
